students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+                    str_donations = "Yes"
+                    break
+                else:
+                    print("Donor is not in donor list. \n")
+        return str_user_input, str_donations
+
+    @staticmethod
+    def user_factor_input():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns factor input to the main function.
+        '''
+        while True:
+            try:
+                int_user_input = float(input(str_name + ", please enter a positive integer. "))
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+                # Verifies that the data type is part of the int class and that the user input a value greater than 0.
+            else:
+                if int_user_input.__class__ == float().__class__ and int_user_input > 0:
+                    break
+                else:
+                    print("Integer selected is not a valid option. "
+                              "Please, enter an integer. \n")
+        return int_user_input
+
+    @staticmethod
+    def user_filter_input():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns filter data to the main function.
+        '''
+        while True:
+            try:
+                str_user_input = input(str_name + ", please enter max to filter for amounts greater than or equal "
+                                                  "to $2,500. \n or min to filter for "
+                                                  "amounts less than $2,500. ").capitalize()
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+            # Verifies that the data type is part of the string class and that the user put in the correct input.
+            else:
+                if str_user_input.__class__ == str().__class__ and (str_user_input == "Max"
+                                                                    or str_user_input == "Min"):
+                    break
+                else:
+                    print("Integer selected is not a valid option. "
+                              "Please, enter max or min. \n")
+        return str_user_input
+
+########### END OF MENU CLASS###############
+
+# Methods defined in the files class are related to opening and renaming files.
+# Modifications to the methods are needed if the files are not for a task priority list.
+
+
+class Files:
+    @staticmethod
+    def write_email(donors):
+        '''
+        Writes emails to the donors and saves them to text files.
+        :param donors: Dictionary that stores all the donor's donation information.
+        :return:
+        '''
+        for donor_name in donors:
+            # Open text file and assign a mode to the file.
+            new_file = open(donor_name + ".txt", mode="w")
```

Comment:
It's a best practice to use a `with` statement for dealing with files and this will automatically close the file for you when you're done with it